pkg/controllers/disruption/suite_test.go

```diff
@@ -467,6 +467,105 @@ var _ = Describe("Pod Eviction Cost", func() {
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{"app": "large-app"},
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion:         "apps/v1",
+						Kind:               "ReplicaSet",
+						Name:               rs.Name,
+						UID:                rs.UID,
+						Controller:         ptr.Bool(true),
+						BlockOwnerDeletion: ptr.Bool(true),
+					},
+				},
+			},
+			// Each 2xlarge has 8 cpu, so each node should fit 2 pods.
+			ResourceRequirements: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceCPU: resource.MustParse("3"),
+				},
+			},
+		})
+
+		nodeClaims, nodes = test.NodeClaimsAndNodes(2, v1beta1.NodeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					v1beta1.NodePoolLabelKey:     nodePool.Name,
+					v1.LabelInstanceTypeStable:   mostExpensiveInstance.Name,
+					v1beta1.CapacityTypeLabelKey: mostExpensiveOffering.CapacityType,
+					v1.LabelTopologyZone:         mostExpensiveOffering.Zone,
+				},
+			},
+			Status: v1beta1.NodeClaimStatus{
+				Allocatable: map[v1.ResourceName]resource.Quantity{
+					v1.ResourceCPU:  resource.MustParse("8"),
+					v1.ResourcePods: resource.MustParse("100"),
+				},
+			},
+		})
+		nodeClaims[0].StatusConditions().MarkTrue(v1beta1.Drifted)
+		nodeClaims[1].StatusConditions().MarkTrue(v1beta1.Drifted)
+
+		ExpectApplied(ctx, env.Client, nodePool, pods[0], nodes[0], nodeClaims[0])
+		ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*v1.Node{nodes[0]}, []*v1beta1.NodeClaim{nodeClaims[0]})
+
+		// Expect the first pod scheduled, and then cordon the node so that the other pod can schedule
+		// to the other node.
+		ExpectManualBinding(ctx, env.Client, pods[0], nodes[0])
+		nodes[0].Spec.Unschedulable = true
+		ExpectApplied(ctx, env.Client, nodes[0])
+
+		ExpectApplied(ctx, env.Client, nodePool, pods[1], nodes[1], nodeClaims[1])
+		ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*v1.Node{nodes[1]}, []*v1beta1.NodeClaim{nodeClaims[1]})
+
+		ExpectManualBinding(ctx, env.Client, pods[1], nodes[1])
+		nodes[1].Spec.Unschedulable = false
+		ExpectApplied(ctx, env.Client, nodes[1])
+
+		// Reconcile twice, but only expect one to be disrupted.
+		// This first reconcile doesn't need an ExpectTriggerVerifyAction as we shouldn't be reaching the consolidation code.
+		// If we do, we'll mess with the nomination window, and the second reconcile will actually succeed to disrupt the second node.
+		ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKey{})
+
+		var wg sync.WaitGroup
+		ExpectTriggerVerifyAction(&wg)
+		ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKey{})
+		wg.Wait()
+
+		nodes = ExpectNodes(ctx, env.Client)
+		Expect(len(nodes)).To(Equal(2))
+		tainted := lo.Filter(nodes, func(n *v1.Node, _ int) bool {
```

Comment:
Should we have an `ExpectTainted` function like we do in our E2Es on the AWS provider?